templates/commands/render.go

```diff
@@ -322,6 +328,35 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// validateAndDefaultInputs validates and sets the input values provided by the user.
+func validateAndDefaultInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var result *multierror.Error
+
+	for _, i := range spec.Inputs {
+		val, ok := flagInputs[i.Name.Val]
+		if !ok || val == "" {
+			if i.Default.Val != "" {
+				flagInputs[i.Name.Val] = i.Default.Val
+			} else if i.Required.Val {
```

Comment:
I'm thinking we should just remove the "Required" field and just say "a field is required if and only if it doesn't have a default." What do you think?

Reply:
I feel like there could be a use case, where a string is optional, and if given you render a certain part of a template. I'd say lets leave it for now.

Reply:
Maybe we can default the default to empty string and remove the required flag, this could achieve the same functionality.

Reply:
I like that...